src/main/java/net/earthcomputer/clientcommands/command/PlayerInfoCommand.java

```diff
@@ -0,0 +1,104 @@
+                                .suggests(((context, builder) -> suggestMatching(MinecraftClient.getInstance().world.getPlayers().stream().map(abstractPlayer -> abstractPlayer.getName().getString()), builder)))
+                                .executes(ctx -> getNameHistory(ctx.getSource(), getString(ctx, "player"))))));
+    }
+
+    private static int getNameHistory(ServerCommandSource source, String player) throws CommandSyntaxException {
+        String uuid;
+        if (player.contains("-")) {
+            uuid = player;
+            if (cacheByUuid.containsKey(uuid)) {
+                sendFeedback(new TranslatableText("commands.cnamehistory.success", player, String.join(", ", cacheByUuid.get(uuid))));
+                return 0;
+            }
+        } else {
+            if (cacheByName.containsKey(player)) {
+                sendFeedback(new TranslatableText("commands.cnamehistory.success", player, String.join(", ", cacheByName.get(player))));
+                return 0;
+            }
+            Optional<String> optional = MinecraftClient.getInstance().world.getPlayers().stream()
+                    .filter(abstractPlayer -> abstractPlayer.getName().getString().equals(player))
+                    .map(Entity::getUuidAsString)
+                    .findFirst();
+            if (optional.isPresent()) {
+                uuid = optional.get();
+            } else {
+                JsonElement result = requestAsync("https://api.mojang.com/users/profiles/minecraft/" + player);
+                if (result instanceof JsonNull) {
+                    throw IO_EXCEPTION.create();
+                }
+                uuid = result.getAsJsonObject().get("id").getAsString();
+            }
+        }
+        JsonArray names = requestAsync("https://api.mojang.com/user/profiles/" + uuid + "/names").getAsJsonArray();
+
+        List<String> stringNames = new ArrayList<>();
+        names.forEach(name -> stringNames.add(name.getAsJsonObject().get("name").getAsString()));
+        cacheByName.put(player, stringNames);
+        cacheByUuid.put(uuid, stringNames);
+        sendFeedback(new TranslatableText("commands.cnamehistory.success", player, String.join(", ", stringNames)));
+        return 0;
+    }
+
+    private static JsonElement requestAsync(String url) throws CommandSyntaxException {
+        CompletableFuture<JsonElement> future = CompletableFuture.supplyAsync(() -> {
+            try {
+                URLConnection request = new URL(url).openConnection();
+                request.connect();
+                JsonElement result = (new JsonParser().parse(new InputStreamReader(request.getInputStream())));
+                if (result == null) {
+                    throw new IOException();
+                }
+                return result;
+            } catch (Exception e) {
+                throw new CompletionException(e);
+            }
+        });
+        try {
+            return future.join();
```

Comment:
This isn't actually async because it causes the main thread to wait until the task is completed. The future should be stored somewhere and queried every tick until it's finished